src/DDTrace/Integrations/ElasticSearch/V1/ElasticSearchIntegration.php

```diff
@@ -0,0 +1,328 @@
+        self::traceNamespaceMethod('ClusterNamespace', 'getSettings');
+        self::traceNamespaceMethod('ClusterNamespace', 'health');
+        self::traceNamespaceMethod('ClusterNamespace', 'pendingTasks');
+        self::traceNamespaceMethod('ClusterNamespace', 'putSettings');
+        self::traceNamespaceMethod('ClusterNamespace', 'reroute');
+        self::traceNamespaceMethod('ClusterNamespace', 'state');
+        self::traceNamespaceMethod('ClusterNamespace', 'stats');
+
+        // NodesNamespace operations
+        self::traceNamespaceMethod('NodesNamespace', 'hotThreads');
+        self::traceNamespaceMethod('NodesNamespace', 'info');
+        self::traceNamespaceMethod('NodesNamespace', 'shutdown');
+        self::traceNamespaceMethod('NodesNamespace', 'stats');
+
+        // Endpoints
+        dd_trace('Elasticsearch\Endpoints\AbstractEndpoint', 'performRequest', function () {
+            $args = func_get_args();
+            $tracer = GlobalTracer::get();
+            $scope = $tracer->startActiveSpan("Elasticsearch.Endpoint.performRequest");
+            $span = $scope->getSpan();
+
+            $span->setTag(Tags\SERVICE_NAME, ElasticSearchIntegration::DEFAULT_SERVICE_NAME);
+            $span->setTag(Tags\SPAN_TYPE, Types\ELASTICSEARCH);
+
+            // PHP 5.4 compatible try-catch-finally
+            $thrown = null;
+            $result = null;
+            try {
+                // Some endpoints can throw exception during getURI() if some parameters are missing, wso
+                // make sure that the uri is read within the try-catch-finally block.
+                $span->setTag(Tags\RESOURCE_NAME, 'performRequest');
+                $span->setTag(Tags\ELASTICSEARCH_URL, $this->getURI());
+                $span->setTag(Tags\ELASTICSEARCH_METHOD, $this->getMethod());
+                if (is_array($this->params)) {
+                    $span->setTag(Tags\ELASTICSEARCH_PARAMS, json_encode($this->params));
+                }
+                if ($this->getMethod() === 'GET' && $body = $this->getBody()) {
+                    $span->setTag(Tags\ELASTICSEARCH_BODY, json_encode($this->getBody()));
+                }
+                $result = call_user_func_array([$this, 'performRequest'], $args);
+            } catch (\Exception $ex) {
+                $thrown = $ex;
+                if ($span instanceof Span) {
+                    $span->setError($ex);
+                }
+            }
+            $scope->close();
+            if ($thrown) {
+                throw $thrown;
+            }
+
+            return $result;
+        });
+    }
+
+    /**
+     * @param string $name
+     */
+    public static function traceClientMethod($name)
+    {
+        $class = 'Elasticsearch\Client';
+        if (!method_exists($class, $name)) {
+            return;
+        }
+
+        dd_trace($class, $name, function () use ($name) {
+            $args = func_get_args();
+            list($params) = extract($args);
```

Comment:
The `extract()` function returns an `int`, not an `array` so I'm not sure what this line is doing. If we remove it and replace `$params` with `$args` on line 188, the tests still pass for me. :)

Reply:
very good catch! I think that I actually should convert to
`list($params) = $args` as different methods can have other stuff into the method signature.
Thoughts? :)

Reply:
Sounds good! If all we need is the first argument, we could even do this and remove `$args` altogether:

```php
list($params) = func_get_args();
```